src/midiEditorCommands.cpp

```diff
@@ -50,6 +50,29 @@ int getItemPPQ(MediaItem* item) {
 	return ppq;
 }
 
+// return the midi editor zoom ratio of the item 
+double getMidiZoomRatio(MediaItem* item) {
+	static const regex re("CFGEDITVIEW -?[0-9.]+ ([0-9.]+) ");
+	const size_t startBuffSize = 16384;
+	const size_t maxBuffSize = 1<<24; // 16 Mib
+	for (size_t buffSize = startBuffSize; buffSize <= maxBuffSize; buffSize *= 2) {
+		string buffer(buffSize, '\0');
+		if (!GetItemStateChunk(item, &buffer.front(), buffSize, false)) {
+			return -1;
+		}
+		smatch match;
+		if (!regex_search(buffer, match, re)) {
+			if (buffer[buffSize-2] != '\0') { //chunk bigger than buffer
```

Comment:
From this, I gather that the CFGEDITVIEW part is towards the end of the chunk (after the take data), rather than being at the start? Or is this out of an abundance of caution? I'm just double checking in case we can go for a much smaller buffer and avoid retries.

Reply:
Unfortunately it is after all the midi events, which are in text format so they take up a lot of space.